tests/Utility/TitleLinkUtilityTest.php

```diff
@@ -0,0 +1,283 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Tests\RST\References;
+
+use Doctrine\RST\Environment;
+use Doctrine\RST\Utility\TitleLinkUtility;
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+
+class TitleLinkUtilityTest extends TestCase
+{
+    /** @var Environment|MockObject */
+    private $environment;
+
+    /** @var TitleLinkUtility */
+    private $titleLinkUtility;
+
+    protected function setUp(): void
+    {
+        $this->environment      = $this->createMock(Environment::class);
+        $this->titleLinkUtility = new TitleLinkUtility($this->environment, 999);
+    }
+
+    public function testBuildEmptyTitleLinksCreateEmptyTocItemArray(): void
+    {
+        $this->titleLinkUtility = new TitleLinkUtility($this->environment, 999);
```

Comment:
This property has already been initialized in `setUp()`

Reply:
but there are tests for different header levels. Not sure how to handle that as i don't know in what order they are executed

Reply:
yes, we pass the desired level in the constructor

Reply:
Ah I think I get it now.

IMO you should use `$this->titleLinkUtility` when the level is 999, and a local variable such as `$titleLinkUtility` otherwise.